exporters/otlp/otlptrace/otlptracegrpc/client_test.go

```diff
@@ -433,3 +442,246 @@ func TestCustomUserAgent(t *testing.T) {
+
+				sm := rm.ScopeMetrics[0]
+				require.Len(t, sm.Metrics, 3)
+
+				assert.Equal(t, instrumentation.Scope{
+					Name:      "go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc",
+					Version:   sdk.Version(),
+					SchemaURL: semconv.SchemaURL,
+				}, sm.Scope)
+
+				metricdatatest.AssertEqual(t, metricdata.Metrics{
+					Name:        otelconv.SDKExporterSpanInflight{}.Name(),
+					Description: otelconv.SDKExporterSpanInflight{}.Description(),
+					Unit:        otelconv.SDKExporterSpanInflight{}.Unit(),
+					Data: metricdata.Sum[int64]{
+						Temporality: metricdata.CumulativeTemporality,
+						DataPoints: []metricdata.DataPoint[int64]{
+							{
+								Attributes: attribute.NewSet(
+									semconv.OTelComponentName("otlp_grpc_span_exporter/1"),
+									semconv.OTelComponentTypeOtlpGRPCSpanExporter,
+									semconv.ServerAddress("127.0.0.1"),
+									semconv.ServerPort(51844),
+								),
+								Value: 0,
+							},
+						},
+					},
+				}, sm.Metrics[0], metricdatatest.IgnoreTimestamp())
+
+				metricdatatest.AssertEqual(t, metricdata.Metrics{
+					Name:        otelconv.SDKExporterSpanExported{}.Name(),
+					Description: otelconv.SDKExporterSpanExported{}.Description(),
+					Unit:        otelconv.SDKExporterSpanExported{}.Unit(),
+					Data: metricdata.Sum[int64]{
+						Temporality: metricdata.CumulativeTemporality,
+						IsMonotonic: true,
+						DataPoints: []metricdata.DataPoint[int64]{
+							{
+								Attributes: attribute.NewSet(
+									semconv.OTelComponentName("otlp_grpc_span_exporter/1"),
+									semconv.OTelComponentTypeOtlpGRPCSpanExporter,
+									semconv.ErrorType(status.Error(codes.Canceled, "")),
+									semconv.ServerAddress("127.0.0.1"),
+									semconv.ServerPort(51844),
+								),
+								Value: 2,
+							},
+						},
+					},
+				}, sm.Metrics[1], metricdatatest.IgnoreTimestamp())
+
+				metricdatatest.AssertEqual(t, metricdata.Metrics{
+					Name:        otelconv.SDKExporterOperationDuration{}.Name(),
+					Description: otelconv.SDKExporterOperationDuration{}.Description(),
+					Unit:        otelconv.SDKExporterOperationDuration{}.Unit(),
+					Data: metricdata.Histogram[float64]{
+						Temporality: metricdata.CumulativeTemporality,
+						DataPoints: []metricdata.HistogramDataPoint[float64]{
+							{
+								Attributes: attribute.NewSet(
+									semconv.OTelComponentName("otlp_grpc_span_exporter/1"),
+									semconv.OTelComponentTypeOtlpGRPCSpanExporter,
+									semconv.ErrorType(status.Error(codes.Canceled, "")),
+									semconv.RPCGRPCStatusCodeCancelled,
+									semconv.ServerAddress("127.0.0.1"),
+									semconv.ServerPort(51844),
+								),
+							},
+						},
+					},
+				}, sm.Metrics[2], metricdatatest.IgnoreTimestamp(), metricdatatest.IgnoreValue())
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.enabled {
+				t.Setenv("OTEL_GO_X_SELF_OBSERVABILITY", "true")
+			}
```

Comment:
There are two test cases and this conditional splits them. They should be made into their own tests to just remove the complexity being added to accommodate everything here.